sklearn/utils/validation.py

```diff
@@ -1418,6 +1419,20 @@ def check_scalar(
             f"Possible values are: {expected_include_boundaries}."
         )
 
+    if max_val is None and include_boundaries in ("right"):
+        raise ValueError(
+            "`include_boundaries`='right' "
+            "without specifying explicitly `max_val` "
+            "is inconsistent."
+        )
+
+    if min_val is None and include_boundaries in ("left"):
+        raise ValueError(
+            "`include_boundaries`='left' "
+            "without specifying explicitly `min_val` "
+            "is inconsistent."
```

Comment:
Could you wrap the error message on 2 lines only and take profit of the 88 characters width.